Lib/defconQt/glyphView.py

```diff
@@ -1500,56 +1864,178 @@ def addHorizontalMetrics(self):
+        if isActiveLayer:
+            representationKey = "defconQt.NoComponentsQPainterPath"
+        else:
+            representationKey = "defconQt.QPainterPath"
+
+        path = glyph.getRepresentation(representationKey)
+        item = QGraphicsPathItem()
+        item.setPath(path)
+
+        pen, brush = self._getDrawingStyleForLayer(layer)
+        if pen: item.setPen(pen)
+        if brush: item.setBrush(brush)
+
+        item.setZValue(zValue)
+
+        self._sceneItems[layer] = item
+        scene.addItem(item)
+        if isActiveLayer:
+            # FIXME: don't like this
+            scene._outlineItem = item
+            self.addComponents()
+        return item;
+
+    def updateActiveLayerPath(self):
+        self.updateLayerPath(self._layer, representationKey="defconQt.NoComponentsQPainterPath")
+
+    def updateLayerPath(self, layer, representationKey="defconQt.QPainterPath"):
+        glyph = layer[self._name]
+        scene = self.scene()
+        path = glyph.getRepresentation(representationKey)
+        self._sceneItems[layer].setPath(path)
+
+    def _getSceneItems(self, key, clear=False):
+        items = self._sceneItems.get(key, None)
+        if items is None:
+            items = []
+            self._sceneItems[key] = items
+        elif clear:
+            scene = self.scene()
+            for item in items:
+                scene.removeItem(item)
+            del items[:]
+        return items
 
     def addComponents(self):
         scene = self.scene()
-        font = self._glyph.getParent()
-        for component in self._glyph.components:
-            glyph = font[component.baseGlyph]
-            path = glyph.getRepresentation("defconQt.QPainterPath")
+        layer = self._layer
+        glyph = self._glyph
+
+        pen, brush = self._getDrawingStyleForLayer(layer, kind='component')
+        components = self._getSceneItems('components', clear=True)
+        for component in glyph.components:
+            if component.baseGlyph not in layer:
+                continue
+            componentGlyph = layer[component.baseGlyph]
+            path = componentGlyph.getRepresentation("defconQt.QPainterPath")
             item = ComponentItem(path, component)
-            item.setZValue(-996)
+            if pen: item.setPen(pen)
+            if brush: item.setBrush(brush)
+            item.setZValue(self._activeLayerZValue)
+            components.append(item)
             scene.addItem(item)
 
     def addAnchors(self):
         scene = self.scene()
+        anchors = self._getSceneItems('anchors', clear=True)
+
         for anchor in self._glyph.anchors:
             item = AnchorItem(anchor, self.transform().m11())
-            item.setZValue(-996)
+            item.setZValue(-992)
+            anchors.append(item)
             scene.addItem(item)
 
     def addPoints(self):
         scene = self.scene()
+
+        ocPoints = self._getSceneItems('ocPoints', clear=True)
```

Comment:
oc doesn't really say whether its onCurves or offCurves... also it would be oCPoints but more descriptive name doesn't hurt